openedx/features/edly/utils.py

```diff
@@ -139,8 +164,35 @@ def update_course_creator_status(request_user, user, set_creator):
     Updates course creator status of a user.
     """
     from course_creators.models import CourseCreator
+    from course_creators.views import update_course_creator_group
     course_creator, __ = CourseCreator.objects.get_or_create(user=user)
-    course_creator.state = CourseCreator.GRANTED if set_creator else CourseCreator.DENIED
+    course_creator.state = CourseCreator.GRANTED if set_creator else CourseCreator.UNREQUESTED
     course_creator.note = 'Course creator user was updated by panel admin {}'.format(request_user.email)
     course_creator.admin = request_user
     course_creator.save()
+    if not set_creator:
+        update_course_creator_group(request_user, user, set_creator)
```

Comment:
should not this statement be called no matter what we have in `set_creator `?

Reply:
There is a post save signal that updates the course creator group in case the status is `GRANTED`